reactor-core/src/main/java/reactor/core/publisher/MonoCompletionStage.java

```diff
@@ -52,7 +60,7 @@ public void subscribe(CoreSubscriber<? super T> actual) {
             @Override
             public void cancel() {
                 super.cancel();
-                if (future instanceof Future) {
+                if (cancel) {
```

Comment:
I believe the `instanceof` check is still required.

Reply:
it is impossible that `cancel == true` and `!future instanceof Future`. The public API allows only `CompletableFuture` to have the `cancel` flag configurable. in all other cases, it is false. This, checking instance of in addition to cancel is true will be overhead

Reply:
check the following code -> https://github.com/reactor/reactor-core/pull/3252/files#diff-e815d2e7a55f9f17006048ff085897e713143567345688c33e9d64b19f911f06R51